dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -54,641 +46,237 @@
-        this(APILocator.getContentletAPI(),
-                VanityUrlServices.getInstance(),
-                APILocator.getLanguageAPI(),
-                APILocator.getUserAPI());
-    }
-
-    @VisibleForTesting
-    public VanityUrlAPIImpl(final ContentletAPI contentletAPI,
-            final VanityUrlServices vanityUrlServices,
-            final LanguageAPI       languageAPI,
-            final UserAPI           userAPI) throws DotDataException {
-
-        this.contentletAPI     = contentletAPI;
-        this.vanityUrlServices = vanityUrlServices;
-        this.languageAPI       = languageAPI;
-        this.userAPI           = userAPI;
-        this.defaultLanguageId = languageAPI.getDefaultLanguage()
-                .getId();
-        this.systemUser        = userAPI.getSystemUser();
 
+  private final Set<Integer> allowedActions = new ImmutableSet.Builder<Integer>().add(200).add(301).add(302).build();
+
+  private static final String SELECT_LIVE_VANITY_URL_INODES =
+      "SELECT cvi.live_inode FROM contentlet c, identifier i, contentlet_version_info cvi, structure s "
+          + " where s.structuretype= 7 and c.structure_inode=s.inode "
+          + " and cvi.live_inode=c.inode and cvi.identifier = i.id  and i.host_inode = ? and cvi.lang =? ";
+
+  public static final String URL_SUFFIX = "/";
+  public static final String LEGACY_CMS_HOME_PAGE = "/cmsHomePage";
+  private final ContentletAPI contentletAPI;
+  private final VanityUrlCache cache;
+  private final LanguageAPI languageAPI;
+
+
+  public VanityUrlAPIImpl()  {
+    this(APILocator.getContentletAPI(),
+        APILocator.getLanguageAPI(), 
+        APILocator.getUserAPI(),
+        CacheLocator.getVanityURLCache());
+  }
+
+  @VisibleForTesting
+  public VanityUrlAPIImpl(final ContentletAPI contentletAPI, 
+      final LanguageAPI languageAPI, 
+      final UserAPI userAPI,
+      final VanityUrlCache cache) {
+    this.contentletAPI = contentletAPI;
+    this.languageAPI = languageAPI;
+    this.cache = cache;
+   
+
+  }
+
+  /**
+   * Searches for all Vanity URLs for a given Site in the system. It goes directly to the database in
+   * order to avoid retrieving data that has not been updated in the ES index yet. This initialization
+   * routine will also add Vanity URLs located under System Host.
+   *
+   * @param siteId String site id
+   * @param languageId Long language id
+   * @return A list of VanityURLs
+   */
+  private void populateVanityURLsCacheBySite(final Host host) {
+    if(host==null || host.getIdentifier()==null) {
+        throw new DotStateException("Host cannot be null. Got:" + host);
     }
-
-    @CloseDBIfOpened
-    @Override
-    public void initializeVanityURLsCache(final User user) {
-        initializeActiveVanityURLsCacheBySiteAndLanguage(null, null);
+    Logger.info(this.getClass(), "Populating Vanity URLS for :" + host.getHostname()); 
+    for (Language lang : languageAPI.getLanguages()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'lang' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)